### ISSUE TITLE
copied web pages isn't close to markdown

### ISSUE THREAD

**Issue by chadmiller** [reddit/reddit#1723]
I sometimes see people compose a post, and because they break some subtle rule ("titles must be somethingsomething"), their post gets deleted. They then copy the rendered text and paste it back into a new form, assuming that it will have preserved their intent for paragraphs and such. 

A \n\n in markdown becomes a paragraph separator, which becomes a \n when copied again. 

paragraph paragraph paragraph becomes paragraphparagraphparagraph.

The output in a web page could preserve their intent through the copy process. A paragraph break could include a new zero-height `<p class="zeroheight"></p>` to be the second newline in copying.

I'm not sure of the details of exposure and copy-action availability, but maybe other markup could survive too.

`[text](link)`

could become

`<a href="link"><span class="hiddenmarkup">[</span>text<span class="hiddenmarkup">](link)</span></a>` or something like that.

**Comment by 13steinj** [comment]
> assuming that it will have preserved their intent for paragraphs and such.

For one thing, reddit likes to follow the markdown spec as closely as possible, so I doubt this would be done.

But even then, it's somewhat pointless to do so. It's adding a ton of data to the rendered text when it's unnecessary. There's no way to properly preserve a few things, even, iirc based on the order that the parser reads the raw markdown. It's also a little ridiculous to assume that one could copy from html and it turn into markdown automagically.

While I do agree it can be annoying, most people I know, including myself, have enough reasoning to press "edit" before copying and pasting into a new post.

**Comment by chadmiller** [comment]
I'm guessing that most people you know are other programmers, then.

**Comment by 13steinj** [comment]
No, actually. Regardless it still breaks the spec and adds extra cruft for little gain, IMO at least.

**Comment by mattjegan** [comment]
@13steinj  can this be closed?

**Comment by 13steinj** [comment]
@mattjegan I'm not an admin nor have any rights on the repository, hence the "Contributor" label and not the "Member" label. I just like my snoo :3
